agent/checks/check_test.go

```diff
@@ -1101,6 +1102,111 @@ func TestCheckTCPPassing(t *testing.T) {
 	tcpServer.Close()
 }
 
+func sendResponse(conn *net.UDPConn, addr *net.UDPAddr) {
+	_, err := conn.WriteToUDP([]byte("healthy"), addr)
+	if err != nil {
+		fmt.Printf("Couldn't send response %v", err)
+	}
+}
+
+func mockUDPServer(network string, port string) net.PacketConn {
+
+	b := make([]byte, 1024)
+
+	addr := `127.0.0.1:` + port
+	udpAddr, err := net.ResolveUDPAddr(network, addr)
+	if err != nil {
+		log.Fatal("Error resolving UDP address: ", err)
+	}
+
+	ser, err := net.ListenUDP("udp", udpAddr)
+	if err != nil {
+		log.Fatal("Error listening UDP: ", err)
+	}
+	for {
+		log.Print("Waiting for UDP message")
+		_, remoteaddr, err := ser.ReadFromUDP(b)
+		log.Printf("Read a message from %v %s \n", remoteaddr, b)
+
+		if err != nil {
+			log.Fatalf("Error reading from UDP")
+		}
+		go sendResponse(ser, remoteaddr)
+	}
+
+}
```

Comment:
I think this should fix the mock server not stopping 
```suggestion
func mockUDPServer(ctx context.Context, network string, port int) {

	b := make([]byte, 1024)

	addr := fmt.Sprintf(`127.0.0.1:%d`, port)
	udpAddr, err := net.ResolveUDPAddr(network, addr)
	if err != nil {
		log.Fatal("Error resolving UDP address: ", err)
	}

	ser, err := net.ListenUDP("udp", udpAddr)
	if err != nil {
		log.Fatal("Error listening UDP: ", err)
	}
	defer ser.Close()
	chClose := make(chan interface{})
	wg := sync.WaitGroup{}

	wg.Add(1)
	go func() {
		for {
			log.Print("Waiting for UDP message")
			_, remoteaddr, err := ser.ReadFromUDP(b)
			log.Printf("Read a message from %v %s \n", remoteaddr, b)

			if err != nil {
				log.Fatalf("Error reading from UDP %s", err.Error())
			}
			sendResponse(ser, remoteaddr)
			select {
			case <-chClose:
				fmt.Println("cancelled")
				wg.Done()
				return
			default:
			}
		}
	}()
	select {
	case <-ctx.Done():
		fmt.Println("cancelled")
		close(chClose)
	}
	wg.Wait()
	return
}
```